ocaml/xapi/storage_smapiv1_migrate.ml

```diff
@@ -108,18 +108,18 @@ let tapdisk_of_attach_info (backend : Storage_interface.backend) =
       let path = blockdevice.Storage_interface.path in
       try
         match Tapctl.of_device (Tapctl.create ()) path with
-        | tapdev, _, _ ->
+        | Some (tapdev, _, _) ->
             Some tapdev
+        | None ->
+            D.debug "Device %s has an unknown driver" path ;
+            None
       with
       | Tapctl.Not_blktap ->
           D.debug "Device %s is not controlled by blktap" path ;
           None
       | Tapctl.Not_a_device ->
           D.debug "%s is not a device" path ;
           None
-      | _ ->
-          D.debug "Device %s has an unknown driver" path ;
-          None
```

Comment:
The try and the match can be combined together, it can be used to protect the user from unexpected exceptions:
```suggestion
      match Tapctl.of_device (Tapctl.create ()) path with
      | Some (tapdev, _, _) ->
          Some tapdev
      | exception Tapctl.Not_blktap ->
          D.debug "Device %s is not controlled by blktap" path ;
          None
      | exeption Tapctl.Not_a_device ->
          D.debug "%s is not a device" path ;
          None
      | exception _ | None ->
          D.debug "Device %s has an unknown driver" path ;
          None
```